### ISSUE TITLE
Propagate fill_in options to Element#set

### ISSUE THREAD

**Issue by sobrinho** [teamcapybara/capybara#1319]
# TL;DR

We need to be able to customise the behaviour of `Node#set` somehow on our driver to trigger blur or not after filling the field, i.e.:

``` ruby
fill_in 'Employee', :name => 'Gabriel Sobrinho', :blur => false
```
# Long Version

Capybara's fill_in extract the :with option and pass it to `Node#set` and we are not able to receive any option from the user.

That's necessary in some specific use cases where the behaviour for fill_in needs to be slight different.

The default behaviour for poltergeist and probably other drivers, as can be seen [here](https://github.com/teampoltergeist/poltergeist/blob/master/lib/capybara/poltergeist/client/agent.coffee#L181-L203), is:
1. Clear input value
2. Trigger focus event
3. Trigger keydown event
4. Append the character
5. Trigger keypress event
6. Trigger keyup event
7. Repeat 3 to 6 for every received character
8. Trigger change event
9. Trigger input event
10. Trigger blur event

We need to be able to be able to receive on poltergeist if the blur event must happen or not.

It's really useful for autocompletes, i.e.: Typeahead and jQuery UI Autocomplete, where the autocomplete options is shown after you stop typing but is still focused on the input.

When the input loses the focus, it get's closed, so we can't chose any option from autocomplete.

There is some recommendations on web to circumvent this "problem" using a custom helper, like this:

``` ruby
  def fill_autocomplete(locator, options)
    value = options.delete(:with)

    field = find(:fillable_field, locator, options)
    field.set(value)

    page.execute_script "$('##{field[:id]}').autocomplete('search')"

    expect(page).to have_selector '.ui-autocomplete .ui-menu-item a'

    page.execute_script "$('.ui-autocomplete .ui-menu-item a:contains(#{value})').trigger('mouseenter').trigger('click')"
  end
```

Receiving the options on `Element#set`, would allow us to do this:

``` ruby
  def fill_autocomplete(locator, options)
    fill_in locator, options.merge(blur: false)

    click_link options[:with]
  end
```

At my first PoC, I changed the capybara from:

``` ruby
module Capybara
  module Node
    module Actions
      def fill_in(locator, options={})
        raise "Must pass a hash containing 'with'" if not options.is_a?(Hash) or not options.has_key?(:with)
        with = options.delete(:with)
        find(:fillable_field, locator, options).set(with)
      end
    end
  end
end

module Capybara
  module Node
    class Element < Base
      def set(value)
        synchronize { base.set(value) }
      end
    end
  end
end
```

To:

``` ruby
module Capybara
  module Node
    module Actions
      def fill_in(locator, options={})
        raise "Must pass a hash containing 'with'" if not options.is_a?(Hash) or not options.has_key?(:with)
        with = options.delete(:with)
        find(:fillable_field, locator, options).set(with, options)
      end
    end
  end
end

module Capybara
  module Node
    class Element < Base
      def set(value, options)
        synchronize { base.set(value, options) }
      end
    end
  end
end
```

And after that I changed signature of `Capybara::Poltergeist::Node#set` to receive 2 arguments, so I received the options and customised the blur event to happen or not.

What do you think, guys?

I think that would be useful for all drivers for all kind of stuff, like [this one](https://github.com/teampoltergeist/poltergeist/issues/488).
